python-lib/wordcloud_visualizer.py

```diff
@@ -122,7 +116,9 @@ def __init__(
 
     def _color_func(self, word, font_size, position, orientation, random_state=None, **kwargs) -> AnyStr:
         """Return the color function used in the wordcloud"""
-        return random.choice(self.color_list)
+        word_hash = zlib.adler32(word.encode("utf-8")) & 0xFFFFFFFF
```

Comment:
In Python3, there shouldn't be a need to `encode`.

Also why adler32 and not the Python builtin `hash?

Reply:
the python builtin hash is not deterministic across different runs of the same job https://stackoverflow.com/a/27955039

Reply:
If I read the thread, I see
> Watch out! I found out the hard way, but adler32's purpose is not for hashing, but for error correction. It has a rather high collision probability. Quite a headache to debug. 

This is not my area of expertise, so I'd advise asking R&D / tech-rescue for advice.

Reply:
collision probability doesn't look like an issue to me, but I'll ask on TR

Reply:
and you need to encode as the wordcloud constructor passes a string to the color_func. Otherwise you end up with the following error: <class 'TypeError'>: a bytes-like object is required, not 'str'

Reply:
kept adler32 because of:
- speed
- availability (python base package)
- doesn't change among python versions (py2 and py3)
- no need for security against collision-bruteforce attacks